src/test/java/io/confluent/connect/jdbc/sink/JdbcDbWriterTest.java

```diff
@@ -71,6 +80,94 @@ private JdbcDbWriter newWriter(Map<String, String> props) {
     return new JdbcDbWriter(config, dialect, dbStructure);
   }
 
+  private JdbcDbWriter newWriterWithMockConnection(Map<String, String> props, Connection mockConnection) {
+    final JdbcSinkConfig config = new JdbcSinkConfig(props);
+    dialect = mock(DatabaseDialect.class);
+    final DbStructure dbStructure = mock(DbStructure.class);
+    return new JdbcDbWriter(config, dialect, dbStructure) {
+      protected CachedConnectionProvider connectionProvider(int maxConnAttempts, long retryBackoff) {
+        CachedConnectionProvider mockConnectionProvider = mock(CachedConnectionProvider.class);
+        when(mockConnectionProvider.getConnection()).thenReturn(mockConnection);
+        return mockConnectionProvider;
+      }
+    };
+  }
+
+  private class MockRollbackException extends SQLException {
+    public MockRollbackException() {
+      super();
+    }
+  }
+
+  @Test
+  public void verifyConnectionRollbackFailed() throws SQLException {
+    SQLException e = connectionRollback(false);
+
+    Throwable[] suppressed = e.getSuppressed();
+    assertEquals(suppressed.length, 1);
+    assertTrue(suppressed[0] instanceof MockRollbackException);
+  }
+
+  @Test
+  public void verifyConnectionRollbackSucceeded() throws SQLException {
+    SQLException e = connectionRollback(true);
+
+    Throwable[] suppressed = e.getSuppressed();
+    assertEquals(suppressed.length, 0);
+  }
+
+  public SQLException connectionRollback(boolean succeedOnRollBack) throws SQLException {
```

Comment:
nit: make the method private and call it something like `assertConnectionRollback` or `verifyConnectionRollback`